app/models/highlight.rb

```diff
@@ -48,8 +48,8 @@ def normalize_ids
 
   protected
 
-  def all_from_scope_and_source
-    Highlight.where(scope_id: scope_id).where(source_id: source_id)
+  def all_mine_from_scope_and_source
+    Highlight.where(user_id: user_id).where(scope_id: scope_id).where(source_id: source_id)
```

Comment:
this could be Highlight.by_user(user_id).where.... (using a by_user scope we already have)